packages/server-wallet/src/wallet/__test__/integration/push-message.test.ts

```diff
@@ -44,40 +45,86 @@ it('stores states contained in the message, in a single channel model', async ()
   expect(signedStates.map(addHash)).toMatchObject(channelsAfter[0].vars);
 });
 
-it('returns a ChannelProposed notification when receiving the first state', async () => {
-  const channelsBefore = await Channel.query().select();
-  expect(channelsBefore).toHaveLength(0);
+const expectResults = async (
+  p: Promise<{channelResults: ChannelResult[]}>,
+  channelResults: Partial<ChannelResult>[]
+): Promise<void> => {
+  await expect(p).resolves.toMatchObject({channelResults});
+  await expect(p.then(data => data.channelResults)).resolves.toHaveLength(channelResults.length);
+};
```

Comment:
I think Jest recommends writing custom matchers instead of helper functions. For example the matcher might be `toHaveChannelResultsLike`... e.g.,

```ts
expect(await wallet.pushMessage(message({signedStates}))).toHaveChannelResultsLike([
    {turnNum: zero, status: 'proposed'},
]);
```

Reply:
I'm happy to do that, but I've tried and failed once before. I'm not very familiar with the typescript required to make that work.

If there is a scaffold in place that lets me just add a `toHaveChannelResultsLike` custom matcher, I would strongly prefer that. Even better if there is autocompletion and type safety!